### PR TITLE
feat: add user graph pages

### PR DESCRIPTION
**Pull request by GrowlingGrizzly**

### CODE REVIEW

File: src/lib/server/functions/getGraphData.ts [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        where: {
+          AND: [
+            { userId: user },
+            { date: { gte: dayjs().subtract(45, "days").toDate() } },
+            { category: category },
+          ],
+        },
+      });
+    } else return "private profile"
+  } else return "invalid user";
```
Comment (tekoh) [inline src/lib/server/functions/getGraphData.ts:29]: Can clean this up by doing the ‘invalid’ checks first

File: src/lib/server/functions/getGraphData.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    userItemCounts.set(dayjs(item.date).format("YYYY-MM-DD"), Number(item.value));
+  }
+
+  const graphData: ChartConfiguration = {
+    type: "line",
+    data: {
+      labels: [],
+      datasets: [
+        {
+          yAxisID: "y2",
```
Comment (tekoh) [inline src/lib/server/functions/getGraphData.ts:45]: Any reason why it’s not y1?

File: src/lib/server/functions/getItemGraphData.ts
Comment (tekoh) [inline src/lib/server/functions/getItemGraphData.ts:1]: Why is this function needed if you have the other one where you can fetch any category

**Review by tekoh — CHANGES_REQUESTED** [review]

File: src/lib/server/functions/getGraphData.ts (hunk 2 above)
Comment (GrowlingGrizzly) [reply src/lib/server/functions/getGraphData.ts:45]: Whenever it's y1 it shows money of the left, however  it doesnt show when hovering over and looks kind of strange with it all zeros on the right side. I don't want to have to redo the entire graph, but if you want to make it look better you can remove the left side numbers and have y1 and y2 just control whether dollar signs show up on the right

File: src/lib/server/functions/getItemGraphData.ts
Comment (GrowlingGrizzly) [reply src/lib/server/functions/getItemGraphData.ts:1]: removed

File: src/lib/server/functions/getGraphData.ts (hunk 1 above)
Comment (GrowlingGrizzly) [reply src/lib/server/functions/getGraphData.ts]: cleaned

File: src/lib/server/functions/getGraphData.ts (hunk 2 above)
Comment (GrowlingGrizzly) [reply src/lib/server/functions/getGraphData.ts:48]: Screenshot 2023-07-16 at 9 43 25 AM
example

**Pull request merged**